lib/puppet/provider/certmonger_certificate/certmonger_certificate.rb

```diff
@@ -75,6 +75,16 @@ def self.parse_cert_list(list_output)
             cn_match = line.match(%r{subject: .*CN=(.*?)(?:,.*|$)})
             current_cert[:hostname] = cn_match[1]
           end
+        when %r{^\s+issuer: .*}
+          # FIXME(owalsh): This is also hacky! Use an actual library to parse
+          # the issuer DN.
+          subj_match = line.match(%r{issuer: (.*)})
+          if subj_match[1].empty?
+            current_cert[:issuer] = ''
+          else
+            cn_match = line.match(%r{issuer: .*CN=(.*?)(?:,.*|$)})
```

Comment:
Uhm... What would this match?

Reply:
Bah, I mistakenly thought that the issuer CN always matched the issuer name. Didn't realised that I just happened to create the sub-CA this way in FreeIPA

Reply:
That seems to be the case. I suggest you just match the whole issuer string.

Reply:
nope, that won't work either. certmonger accepts the issuer name when creating the request but reports the issuer_dn when querying.

Looks like the defined type hit this too, and requires both the issuer name and issuer DN: https://github.com/saltedsignal/puppet-certmonger/blob/master/manifests/request_ipa_cert.pp#L133